app/main/posts/views/filters/filters-dropdown.directive.js

```diff
@@ -44,7 +42,8 @@ function FiltersDropdown(PostFilters, ModalService, $rootScope, _, $location, Sa
         $scope.clearFilters = function () {
             if (PostFilters.getMode() === 'savedsearch' && PostFilters.getModeId()) {
                 PostFilters.setMode('all');
-                var viewParam = $scope.$resolve.$transition$.params().view ? $scope.$resolve.$transition$.params().view : 'data';
+                // @uirouter-refactor this is very wrong.
+                var viewParam = $scope.view ? $scope.view : 'data';
```

Comment:
this is check map vs data view?
Can you inject $state and just check `$state.current.name`? or $state.params?

Reply:
:( I can't. It fails when you try to inject it in a directive that has a link and no controller

Reply:
it used to be possible to just call $stateParams, which is still a ridiculous thing to do because it's deprecated. So not sure